src/controllers/auth.controller.js

```
@@ -0,0 +1,127 @@
+import { authService } from '../services/auth.service.js';
+import bcrypt from 'bcrypt';
+import {
+  normalize,
+  validateEmail,
+  validatePassword,
+} from '../utils/validate.js';
+import { mailer } from '../utils/mailer.js';
+import { jwt } from '../utils/jwt.js';
+
+const register = async (req, res) => {
+  const { password, name, email } = req.body;
+
+  if (!password || !name || !email) {
+    return res.status(401).json({ message: 'Invalid credentials' });
+  }
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (Object.values(errors).some((error) => error)) {
+    return res.status(400).json({
+      errors,
+      message: 'Validation error',
+    });
+  }
+
+  const existingUser = await authService.getByEmail(email);
+
+  if (existingUser) {
+    return res.status(400).json({
+      errors: { email: 'Email is already taken' },
+      message: 'Validation error',
+    });
+  }
+
+  const saltRounds = 10;
+  const hashedPassword = await bcrypt.hash(password, saltRounds);
+  const activationToken = bcrypt.genSaltSync(1);
+
+  const user = await authService.create({
+    name,
+    email,
+    password: hashedPassword,
+    activationToken,
+  });
+
+  await mailer.sendActivationLink(email, activationToken);
+
+  return res.status(201).json(normalize(user));
+};
+
+const activation = async (req, res) => {
+  const { email } = req.params;
+  const token = req.params.token;
+
+  const user = await authService.getByEmail(email);
+
+  if (!user || user.activationToken !== token) {
+    return res.status(404);
```

Error: The activation endpoint returns only a status code (res.status(404)) without a response body. This is not standard and may cause issues for clients expecting a JSON response. You should return a JSON error message, e.g., 'return res.status(404).json({ message: "Invalid activation link" });'.